packages/block-editor/src/components/list-view/block-select-button.js

```diff
@@ -87,7 +193,24 @@ function ListViewBlockSelectButton(
 					spacing={ 1 }
 				>
 					<span className="block-editor-list-view-block-select-button__title">
-						<Truncate ellipsizeMode="auto">{ blockTitle }</Truncate>
+						{ supportsBlockNaming && labelEditingMode ? (
+							<InputControl
```

Comment:
I'm not clear on the a11y of having a random input inside a button element. If this is not legit or useable for assistive tech then what other patterns could we explore here?

Reply:
This does not work at all. 👎 We will need to break out the 2 elements somehow and conditionally render the input.

Reply:
Would it help at all if the `InputControl` was technically in a Popover, that's positioned so that visually it appears to be displayed over the `Button`? I was wondering if that might help the semantics here.

Reply:
Just another thought, if the `Popover` idea is too heavy-handed, another alternative would be take the `aria-hidden={ true }` line from the parent Button and try moving it down to the `Truncate` component instead, so that we're just hiding the label rather than the entire button.

Reply:
My feeling is it should be a sibling of the button.

Possibly moving the implementation up to `ListViewBlockContents` with a ternary between rendering the button and the input, which is not too dissimilar to how there's already a ternary.

The issue is that you then need a way to render the expander, block icon, anchor and lock icon with the exact same styling between normal and editing modes, but it's not impossible to solve either by wrangling styles or making a presentational component that accepts `as={ Button }` or `as={ input }` props.

Reply:
What i'm hearing is - as I suspected - the usage of form elements within a button is not good for a11y. Let's solve that.

> it's not impossible to solve either by wrangling styles or making a presentational component that accepts as={ Button } or as="div" props.

I understand @talldan's suggestion to be that we extract a presentational component that renders _identically_ but can be a button or a div. That way if the node is in "edit" mode the wrapping `<button>` can become a `<div>` and then switched back to a `<button>` when existing "edit" mode.

Reply:
Roughly this when shown normally:
`<ListViewSomethingOrOther as={ button } onClick={ // ... }>{ labelAsText}</ListViewSomethingOrOther>`

and this in edit mode:
`<ListViewSomethingOrOther as={ div }>{ labelAsInput }</ListViewSomethingOrOther>`

I don't know if it's weird to only have the label as the child. WDYT?

The component would actually be almost the same as `ListViewBlockSelectButton` is in trunk, but just a few changes. I don't know what it'd be called as there are already a few components with similar meaning names 😅

Reply:
I took a look at this and I [ended up implementing this without having to create another component](https://github.com/WordPress/gutenberg/pull/42605#issuecomment-1227116597). Not sure whether that's better or not. I think some of the event handlers could be removed when in editing mode so the conditionals might start getting a little overwhelming and a separate component _could_ be easier.